swagger.yaml

```diff
@@ -0,0 +1,333 @@
+        500:
+          $ref: '#/responses/InternalError'
+    put:
+      tags:
+      - "Filter"
+      summary: "Update the state of the job"
+      description: |
+        Update the filter job by providing new properties
+      parameters:
+      - $ref: '#/parameters/filter_job'
+      responses:
+        200:
+          description: "The state has been updated"
+        400:
+          description: "The job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterJobNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}/dimensions:
+    get:
+      tags:
+       - "Filter"
+      summary: "Get all dimensions used by this filter"
+      description: |
+        Return a list of all dimensions which are going to be used to filter on
+      parameters:
+      - $ref: '#/parameters/filter_id'
+      responses:
+        200:
+          description: "A list of dimension URLs"
+          schema:
+            type: array
+            items:
+              type: string
+        404:
+          $ref: '#/responses/FilterJobNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+  /filters/{filter_id}/dimensions/{dimension_name}:
+    parameters:
+      - $ref: '#/parameters/filter_id'
+      - $ref: '#/parameters/dimension_name'
+    post:
+      tags:
+      - "Filter"
+      summary: "Add a dimension to filter on with a list of values"
+      description: "The dimension can only be added into the job if the state is still set to `created` otherwise 400 status code is returned"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/dimension'
+      responses:
+        200:
+          description: "The dimension was created"
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrdimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    delete:
+      tags:
+      - "Filter"
+      summary: "Remove a dimension and any values set within the dimension"
+      description: "Remove a dimension and any values set within the dimension"
+      responses:
+        200:
+          description: "The dimension was removed"
+        400:
+          description: "The filter job has been locked as it has been `submitted` to be processed"
+        404:
+          $ref: '#/responses/FilterOrdimensionNotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    get:
+      tags:
+      - "Filter"
+      summary: "Get all values from a dimension which have been set"
+      description: "Get a list of all values which will be used to filter the dimension"
+      responses:
+        200:
+          description: "A list of all values for a dimension was returned"
+          schema:
+            $ref: '#/definitions/Dimension'
+        400:
```

Comment:
Would we respond with status 400 on a GET request for locking a submitted filter?

Also a 400 could be returned if the `filter_id` does not exist and a 404 if the `dimension_name` does not exist.